src/core/api/emit_seek_events.ts

```diff
@@ -49,9 +48,15 @@ export default function emitSeekEvents(
       return EMPTY;
     }
 
-    const isSeeking$ = observableFromEvent(mediaElement, "seeking")
-      .pipe(mapTo("seeking" as const));
-
+    const isSeeking$ = observableFromEvent(mediaElement, "seeking").pipe(
+      switchMapTo(
+        clock$.pipe(
+          mergeMap((tick: IClockTick) => {
+            return tick.stalled !== null && tick.stalled.reason !== "internal-seek" ?
```

Comment:
This is dangerous because both the `clock$` Observable and that one listen to the `"seeking"` event from the media element.
Here we have a risk of a race condition where this observable's callback would be called before the `clock$`'s one thus `stalled.reason` not being set to "internal-seek" yet.

A possible resolution of that problem would just to be relying on the clock$ and not directly on the "seeking" event anymore.
But as `stalled.reason` is only set to `"seeking"` when there's no data around we also have to take care of emitting here even if `stalled` is `null`. Maybe we can rely on the clock's state attribute, which contains the name of the event that triggered the clock's tick.